pkg/controller/worker/genericactuator/actuator.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genericactuator
+
+import (
+	"context"
+
+	"github.com/gardener/gardener-extensions/pkg/controller/worker"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+type genericActuator struct {
+	logger   logr.Logger
+	actuator Actuator
```

Comment:
When I first looked at the PR I got a bit confused by the actuator in actuator design. I guess it's mainly because this `Actuator` with its `Context` function has different characteristics compared to the other actuators in our extensions.

I'd recommend to rename the `Actuator` interface to `DelegateFactory` as well as the `Context` interface to `WorkerDelegate`.
WDYT?

Reply:
Sure, sounds good! Will do.